src/parser.js

```diff
@@ -678,7 +686,7 @@ export default class Parser extends Stream {
 
                 this.manifest.dateRanges[index].endDate = new Date(newDateInSeconds);
               }
-              if (dateRange && !this.manifest.dateTimeString) {
+              if (dateRange && !('programDateTimeString' in currentUri)) {
```

Comment:
I guess we can simply check `this.lastProgramDateTime !== null`.

Side question:
In this particular case, we check that at least one program date time **before** must be before the first date range tag. 
I would assume that the program date time tag can be after the date range tag (meaning it is presented in a playlist..., in this case, it will increase complexity for us :)